server/routes/api/fileOperations.js

```diff
@@ -88,12 +88,43 @@ router.post("fileOperations.redirect", auth(), async (ctx) => {
   authorize(user, fileOp.type, team);
 
   if (fileOp.state !== "complete") {
-    throw new ValidationError("file operation is not complete yet");
+    throw new ValidationError(`${fileOp.type} is not complete yet`);
   }
 
   const accessUrl = await getSignedUrl(fileOp.key);
 
   ctx.redirect(accessUrl);
 });
 
+router.post("fileOperations.delete", auth(), async (ctx) => {
+  const { id } = ctx.body;
+  ctx.assertUuid(id, "id is required");
+
+  const user = ctx.state.user;
+  const team = await Team.findByPk(user.teamId);
+  const fileOp = await FileOperation.findByPk(id);
+
+  if (!fileOp) {
+    throw new NotFoundError();
+  }
+
+  authorize(user, fileOp.type, team);
+
+  if (fileOp.state === "expired") {
+    throw new ValidationError(`${fileOp.type} is already expired`);
+  }
+
+  await fileOp.expire();
```

Comment:
We need to actually delete the model, probably wrap this in a command and a transaction too. Otherwise after clicking delete and refreshing…

![image](https://user-images.githubusercontent.com/380914/135790765-0c13ab8c-21a3-4825-8c82-c418735e5343.png)

lmk if you don't have time to make the changes, np if so

Reply:
I see. I will do it in the evening 👍🏻